bundle-httpcache/src/main/java/com/adobe/acs/commons/httpcache/config/impl/GroupHttpCacheConfigExtension.java

```diff
@@ -44,14 +70,125 @@ public boolean accepts(SlingHttpServletRequest request, HttpCacheConfig cacheCon
                         .getResourceResolver().adaptTo(User.class));
 
                 // At least one of the group in config should match.
-                boolean isGroupMatchFound = CollectionUtils.containsAny(customConfigAttributes, requestUserGroupNames);
+                boolean isGroupMatchFound = CollectionUtils.containsAny(userGroups, requestUserGroupNames);
                 if (!isGroupMatchFound) {
-                    log.debug("Group didn't match and hence rejecting the cache config.");
+                    log.trace("Group didn't match and hence rejecting the cache config.");
+                } else {
+                    if (log.isTraceEnabled()) {
+                        log.trace("GroupHttpCacheConfigExtension accepts request [ {} ]", request.getRequestURI());
+                    }
                 }
                 return isGroupMatchFound;
             } catch (RepositoryException e) {
-                throw new HttpCacheReposityAccessException("Unable to access group information of request user.", e);
+                throw new HttpCacheRepositoryAccessException("Unable to access group information of request user.", e);
             }
         }
     }
+
+    //-------------------------<CacheKeyFactory methods>
```

Comment:
Moved GroupCacheKeyFactory into this same class; Allows sharing of the userGroups OSGi Property